index.js

```diff
@@ -3,28 +3,42 @@ var _ = require('lodash');
 var {
   commit,
   set,
-  stringify,
+  get,
 } = require('./lib');
 
 
 function JMON(data) {
+  var type = Object.prototype.toString.call(data);
+
+  var isObject = (
+    type === '[object Object]' ||
+    type === '[object Array]'
+  );
+
+  if (!isObject){
+    this.initialData = data;
+    this.data = data;
+    return;
```

Comment:
As we are creating a new JMOn object for keys of an object. if we throw a error on this point also we would be returning a empty JMON object instead of data. And returning data wont help as we are using new operator. 

we could  also solve this problem by checking if value for any key is object or not while doing recursion. 
 
Here i am storing data rather than returning anything for consistent approach. By this approach we will have all the information reagrding any keys at every level and we could tackle those things in  seperate functions .

Reply:
@bibekluitel 
Actually after going through this code.
I feel its not recursion, everytime you create `new JMON` it creates a new instance of that.
So it does not expects a return value.

Also, since you are only returning empty return, assigning data does not makes sense, its just a error for us.

Reply:
@arshadkazmi42  
Ya once it creates a new instance of the JMON, old data will be lost from parent data. that is why i have stored the variable here. Else we have to prevent creating new JMON instance for primitive data types.